torch/csrc/jit/source_location.h

```diff
@@ -14,13 +16,22 @@ namespace torch { namespace jit {
 struct SourceLocation {
   virtual ~SourceLocation() = default;
   virtual void highlight(std::ostream & out) const = 0;
-  void wrapAndRethrowException(const std::exception & e, const std::string & additional = "") {
-    std::stringstream msg;
+
+  void wrapException(std::ostream& msg, const std::exception & e, const std::string & additional = "") {
     msg << "\n" << e.what() << ":\n";
     if(!additional.empty()) {
       msg << additional << ":\n";
     }
     highlight(msg);
+  }
+  void wrapAndRethrowException(const JITException & e, const std::string & additional = "") {
```

Comment:
same here: two copies of the same exception.

Reply:
Since the function is re-throwing a new exception - JITException/std::runtime_error - don't you need two separate functions to differentiate the newly instantiated exception ?

Reply:
ok, I see what you are doing here, but it is a confusing organization: It overloads an operator based on a static (JITException vs exception) subtype. And the requires a static check and cast to dispatch to that subtype. It would be better if the only function in SourceLocation were `void wrapException(std::ostream& msg, const std::exception & e, const std::string & additional = "")` and then the remaining logic could go in interpreter.cpp